Makefile

```diff
@@ -229,6 +229,8 @@ helm-chart-manifests: generate
 	# Ensure that what we're about to try to replace actually exists (if it doesn't we want to fail)
 	grep -E $(KUBE_RBAC_PROXY) ./charts/azure-service-operator/templates/generated/*_deployment_*
 	perl -pi -e s,$(KUBE_RBAC_PROXY),"{{ .Values.image.kubeRBACProxy }}",g ./charts/azure-service-operator/templates/generated/*_deployment_*
+	# replace hard coded cert-manager version with templating
+	perl -pi -e s,cert-manager.io/v1alpha2,"{{ .Values.certManagerResourcesAPIVersion }}",g ./charts/azure-service-operator/templates/generated/*cert-manager.io*
```

Comment:
I think here the best practice would be to match with `cert-manager.io/<REGEX_MATCH>` and then replace. The reason behind is that it'll reduce an extra step and complexity to change in places in the future if we're changing versions.

Reply:
Fixed, although changed to only replace `apiVersion: cert-manager.io/.*` just in case there's something else in the files that matches `cert-manager.io/` that we don't want to replace (just a little extra safety that we're only replacing the APIVersion)